lmdeploy/pytorch/models/internvl.py

```diff
@@ -1,17 +1,310 @@
+                                         self.embed_dim,
+                                         bias=True,
+                                         quant_config=quantization_config,
+                                         dtype=dtype,
+                                         device=device,
+                                         is_tp=True)
+
+    def forward(self, hidden_states):
+        """forward."""
+
+        # qkv proj
+        qkv_states = self.qkv(hidden_states)
+        q, k, v = self.qkv.split_qkv(qkv_states)
+
+        if self.qk_normalization:
+            q_shape = q.shape
+            q = self.q_norm(q.flatten(-2, -1)).view(q_shape)
+            k = self.k_norm(k.flatten(-2, -1)).view(q_shape)
+
+        q = q.transpose(1, 2)
+        k = k.transpose(1, 2)
+        v = v.transpose(1, 2)
+
+        attn_output = F.scaled_dot_product_attention(q, k, v, scale=self.scale)
+
+        # o proj
+        attn_output = attn_output.transpose(1, 2)
+        attn_output = attn_output.flatten(-2, -1)
+        attn_output = self.proj(attn_output)
+        return attn_output
+
+
+class InternMLP(nn.Module):
+    """intern vl mlp."""
+
+    def __init__(self,
+                 config: PretrainedConfig,
+                 dtype: torch.dtype = None,
+                 device: torch.device = None):
+        super().__init__()
+        from transformers.activations import ACT2FN
+        self.config = config
+        quantization_config = getattr(config, 'quantization_config', None)
+        self.act = ACT2FN[config.hidden_act]
+
+        self.fc1 = build_colwise_linear(
+            config.hidden_size,
+            config.intermediate_size,
+            bias=True,
+            dtype=dtype,
+            device=device,
+            quant_config=quantization_config,
+            is_tp=True,
+        )
+
+        self.fc2 = build_rowwise_linear(config.intermediate_size,
+                                        config.hidden_size,
+                                        bias=True,
+                                        quant_config=quantization_config,
+                                        dtype=dtype,
+                                        device=device,
+                                        is_tp=True)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states = self.fc1(hidden_states)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.fc2(hidden_states)
+        return hidden_states
+
+
+class InternVisionEncoderLayer(nn.Module):
+    """intern vision encoder layer."""
+
+    def __init__(self,
+                 config: PretrainedConfig,
+                 dtype: torch.dtype = None,
+                 device: torch.device = None):
+        super().__init__()
+        self.config = config
+        self.embed_dim = config.hidden_size
+        self.intermediate_size = config.intermediate_size
+        self.norm_type = config.norm_type
```

Comment:
here may use getattr `norm_type` since this attr does not exists for models like [ChemVLM](https://huggingface.co/AI4Chem/ChemVLM-26B/blob/5d2a25396311101ab886176ad96ddcd586458790/modeling_intern_vit.py#L273)


```
[rank0]:   File "/lmdeploy/lmdeploy/pytorch/models/internvl.py", line 220, in __init__
[rank0]:     self.norm_type = config.norm_type
[rank0]:   File "/opt/py3/lib/python3.10/site-packages/transformers/configuration_utils.py", line 205, in __getattribute__
[rank0]:     return super().__getattribute__(key)
[rank0]: AttributeError: 'InternVisionConfig' object has no attribute 'norm_type'
```